gneiss/plot/_dendrogram.py

```diff
@@ -0,0 +1,235 @@
+
+Note: This is directly ported from pycogent.
+"""
+
+# Future:
+#  - font styles
+#  - orientation switch
+# Layout gets more complicated for rooted tree styles if dy is allowed to vary,
+# and constant-y is suitable for placing alongside a sequence alignment anyway.
+from skbio import TreeNode
+import pandas as pd
+import numpy
+import abc
+
+
+class Dendrogram(TreeNode):
+    # One of these for each tree edge.  Extra attributes:
+    #    depth - distance from root to bottom of edge
+    #    height - max distance from a decendant leaf to top of edge
+    #    width - number of decendant leaves
+    # note these are named tree-wise, not geometricaly, so think
+    # of a vertical tree (for this part anyway)
+    #
+    #   x1, y1, x2, y2 - coordinates
+    # these are horizontal / vertical as you would expect
+    #
+    # The algorithm is split into 4 passes over the tree for easier
+    # code reuse - vertical drawing, new tree styles, new graphics
+    # libraries etc.
+
+    aspect_distorts_lengths = True
+
+    def __init__(self, use_lengths=True, **kwargs):
+        """ Constructs a Dendrogram object for visualization.
+
+        Parameters
+        ----------
+        use_lengths: bool
+            Specifies if the branch lengths should be included in the
+            resulting visualization (default True).
+        """
+        super().__init__(**kwargs)
+        self.use_lengths_default = use_lengths
+
+    def _cache_ntips(self):
+        for n in self.postorder(include_self=True):
+            if n.is_tip():
+                n._n_tips = 1
+            else:
+                n._n_tips = sum(c._n_tips for c in n.children)
+
+    def coords(self, height, width):
+        """ Returns coordinates of nodes to be rendered in plot.
+
+        Parameters
+        ----------
+        height : int
+            The height of the canvas.
+        width : int
+            The width of the canvas.
+
+        Returns
+        -------
+        pd.DataFrame
+            index : str
+                name of node
+            x : float
+                x-coordinate of point
+            y : float
+                y-coordinate of point
+            left : str
+                name of left child node
+            right : str
+                name of right child node
+        """
+        self.rescale(width, height)
+        result = {}
+        for node in self.postorder(include_self=True):
+            children = {'child%d' % i: n.name
+                        for i, n in enumerate(node.children)}
+            coords = {'x': node.x2, 'y': node.y2}
+            is_tip = {'is_tip': node.is_tip()}
+            result[node.name] = {**coords, **children, **is_tip}
+        result = pd.DataFrame(result).T
+
+        # reorder so that x and y are first
+        cols = list(result)
+        cols.insert(0, cols.pop(cols.index('y')))
+        cols.insert(0, cols.pop(cols.index('x')))
+        result = result.ix[:, cols]
+        return result
```

Comment:
I haven't tested it, but what about this one liner?
return results.loc[:,['x','y'] + list(set(result.columns) - set(['x','y']))]